drivers/exec/handle.go

```diff
@@ -65,6 +66,11 @@ func (h *taskHandle) run() {
 	defer h.stateLock.Unlock()
 
 	if err != nil {
+		h.logger.Warn("failed to wait for process", "pid", h.pid, "error", err)
+		if cErr := executor.CleanupExecutor(h.logger, h.execCleanupHandle); cErr != nil {
```

Comment:
Note that I'm avoiding starting an executor plugin just to perform cleanup task.  I'm not sure what the value of it, considering the call is relatively quick; but we can move spin up an ephemeral executor if we want to be consistent.